crates/biome_fs/src/fs/os.rs

```diff
@@ -234,6 +235,12 @@ impl<'scope> OsTraversalScope<'scope> {
 
 impl<'scope> TraversalScope<'scope> for OsTraversalScope<'scope> {
     fn evaluate(&self, ctx: &'scope dyn TraversalContext, path: Utf8PathBuf) {
+        let path = match std::path::Path::new(&path).absolutize() {
+            Ok(std::borrow::Cow::Owned(absolutized)) => {
+                Utf8PathBuf::from_path_buf(absolutized).unwrap_or(path)
```

Comment:
Using `unwrap_or()` may make it very hard to debug if there is ever an issue with an absolute path, since it means we’ll unexpectedly pass a relative path instead. I’d just use `.expect()` here.